### PR TITLE
Feat edit and delete

### PR DESCRIPTION
**Pull request by ankit9015**
CRUD functionality updated
1. Edit function added for notes cards
2. Modal Editor 
3. Delete/Reset function added

### CODE REVIEW

**Review by dikshadhote — APPROVED** [review]
- In Edit functionality, don't give archive  or delete .That can be done once the card is created
-  Give option to close color pallete and  label container 
image
-  for the Save icon , give title attribute as save or changed icon to save button. The user is not able to find where to look for save button
image
-increase icon sized when card is created 
image

File: src/components/LabelButton/LabelButton.jsx [hunk 1]
```diff
@@ -1,12 +1,15 @@
...
 import { MdOutlineLabel } from "../../utils/icons/icons";
 import "./label-button.css";
 
-function LabelButton({ noteData, setNoteData }) {
+function LabelButton({ setModal, modalState }) {
   const [newLabel, setNewLabel] = useState("");
   const [showDropdown, setShowDropdown] = useState(false);
-  console.log(noteData);
+  const { notesState, notesDispatch, noteLabels, setNoteLabel } = useNotes();
+  // const isInTags = (label) => notesState.tags.includes(label);
```
Comment (rkapoor10) [inline src/components/LabelButton/LabelButton.jsx:11]: remove comments if not needed

File: src/components/LabelButton/LabelButton.jsx [hunk 2]
```diff
@@ -32,18 +35,31 @@ function LabelButton({ noteData, setNoteData }) {
...
           </button>
         </div>
+        <ul>
+          {noteLabels.map((label) => (
+            <label key={label}>
+              <input
+                type="checkbox"
+                name={label}
+                // checked={isInTags(label)}
+                // onClick={labelCheckToggle(label)}
```
Comment (rkapoor10) [inline src/components/LabelButton/LabelButton.jsx:57]: here also

File: src/components/LabelButton/LabelButton.jsx [hunk 3]
```diff
@@ -32,18 +35,31 @@ function LabelButton({ noteData, setNoteData }) {
...
-                  newLabel !== ""
-                    ? [...noteData.tags, newLabel]
-                    : [...noteData.tags],
-              });
+              modalState
+                ? setModal({
+                    ...modalState,
+                    tags: [...modalState.tags, newLabel],
+                  })
+                : notesDispatch({ type: "UPDATE-TAGS", payload: newLabel });
```
Comment (rkapoor10) [inline src/components/LabelButton/LabelButton.jsx:43]: label or tags use a single name for better understanding

File: src/context/NotesContext/NotesContext.js [hunk 4]
```diff
@@ -1,13 +1,118 @@
...
+  useReducer,
+} from "react";
+import addNoteService from "../../service/Notes/addNoteService";
+
+import getNotesService from "../../service/Notes/getNotesService";
+import {
+  updateNotePinService,
+  updateNoteService,
+} from "../../service/Notes/updateNoteService";
+import deleteNotesService from "../../service/Notes/deleteNotesService";
```
Comment (rkapoor10) [inline src/context/NotesContext/NotesContext.js:15]: liked the use of services here

File: src/service/Notes/updateNoteService.js [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+      {
+        headers: {
+          authorization: tokenValue,
+        },
+      }
+    );
+    return response;
+  } catch (error) {
+    console.log(error);
+    error;
```
Comment (rkapoor10) [inline src/service/Notes/updateNoteService.js]: why `error` in line no 17

File: src/service/Notes/updateNoteService.js [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
```
Comment (rkapoor10) [inline src/service/Notes/updateNoteService.js]: here also

**Review by rkapoor10 — APPROVED** [review]
The code is well structured and looks good to merge. 
 - Liked the use of services from an external folder

File: src/components/LabelButton/LabelButton.jsx (hunk 3 above)
Comment (ankit9015) [reply src/components/LabelButton/LabelButton.jsx:43]: OK

File: src/service/Notes/updateNoteService.js (hunk 6 above)
Comment (ankit9015) [reply src/service/Notes/updateNoteService.js]: Thanks for pointing this

**Pull request merged**